packages/rabbitmq/src/TxSubmitWorker.ts

```diff
@@ -254,29 +258,80 @@ export class TxSubmitWorker {
   /**
    * Submit a tx to the provider and ack (or nack) the message
    *
-   * @param {Message} message the message containing the transaction
+   * @param message the message containing the transaction
    */
   private async submitTx(message: Message) {
+    const counter = ++this.#counter;
+    let errorDescription: ErrorDescription | null = null;
+    let txId = '';
+
     try {
-      const counter = ++this.#counter;
       const { content } = message;
+      const txBody = new Uint8Array(content);
+
+      // Register the handling of current transaction
+      txId = txBodyToId(txBody, this.#config.dummyTxId);
 
-      this.#dependencies.logger!.info(`${moduleName}: submitting tx`);
-      this.#dependencies.logger!.debug(`${moduleName}: tx ${counter} dump:`, content.toString('hex'));
-      await this.#dependencies.txSubmitProvider.submitTx(new Uint8Array(content));
+      this.#dependencies.logger!.info(`${moduleName}: submitting tx #${counter} id: ${txId}`);
+      this.#dependencies.logger!.debug(`${moduleName}: tx #${counter} dump:`, content.toString('hex'));
+      await this.#dependencies.txSubmitProvider.submitTx(txBody);
 
-      this.#dependencies.logger!.debug(`${moduleName}: ACKing RabbitMQ message ${counter}`);
+      this.#dependencies.logger!.debug(`${moduleName}: ACKing RabbitMQ message #${counter}`);
       this.#channel?.ack(message);
     } catch (error) {
+      errorDescription = await this.submitTxErrorHandler(error, counter, message);
+    } finally {
```

Comment:
Since it's checking for `!errorDescription` it may be cleaner to just put everything from `finally` block into `catch`

Reply:
The code piece inside the `finally` block needs to run also in case of success: can't be simply moved inside the catch `block`; we could find some other approach...
The response must be sent both in case of success and of not retriable error, the content of the result message should change depending on the success/error case.

Reply:
Oh I see, ok thanks